Rubberduck.CodeAnalysis/Inspections/Concrete/ExcessiveInterfaceMembersInspection.cs

```diff
@@ -0,0 +1,96 @@
+    /// </summary>
+    /// <why>
+    /// Interfaces should not be designed to continually grow new members; we should be keeping them small, specific, and specialized.
+    /// </why>
+    /// <example hasResult="false">
+    /// <module name="MyModule" type="Class Module">
+    /// <![CDATA[
+    /// Option Explicit
+    /// '@Interface
+    ///
+    /// Public Sub DoSomething()
+    /// 
+    /// End Sub
+    /// ]]>
+    /// </module>
+    /// </example>
+    /// <example hasResult="true">
+    /// <module name="MyModule" type="Class Module">
+    /// <![CDATA[
+    /// Option Explicit
+    /// '@Interface
+    ///
+    /// Public Sub DoSomething1()
+    /// 
+    /// End Sub
+    /// 
+    /// Public Sub DoSomething2()
+    /// 
+    /// End Sub
+    /// 
+    /// '...
+    /// 
+    /// Public Sub DoSomethingNGreaterThanMaxPublicMemberCount()
+    ///  
+    /// End Sub
+    /// ]]>
+    /// </module>
+    /// </example>
+    /// 
+
+    internal sealed class ExcessiveInterfaceMembersInspection : DeclarationInspectionBase 
+    {
+        private static IEnumerable<Declaration> _members;
+        private static int ExcessiveMemberCount = 10; //todo: make setting rather than constant
+        
+        public ExcessiveInterfaceMembersInspection(IDeclarationFinderProvider declarationFinderProvider)
+            : base(declarationFinderProvider, DeclarationType.ClassModule) 
+        {}
+
+        protected override bool IsResultDeclaration(Declaration declaration, DeclarationFinder finder) => IsInterfaceDeclaration(declaration) ? HasExcessiveMembers((ClassModuleDeclaration)declaration) : false;
+
+        private static bool IsInterfaceDeclaration(Declaration declaration) => declaration is ClassModuleDeclaration classModule ? classModule.IsInterface : false;
+
+        private static bool HasExcessiveMembers(ClassModuleDeclaration declaration)
+        {
+            _members = declaration.Members.Where(member => { int acc = (int)member.Accessibility; return acc >= 3 && acc <= 5; }); //get rid of non-public members
+            _members = _members.Where(member => !(member.DeclarationType == DeclarationType.Event)); //get rid of public member types that are not part of VBA interface
+            _members = _members.Where(member => member.DeclarationType == DeclarationType.PropertyGet ? NoMatchingSetter(member) : true); //get rid of PropertyGet statements for which there is a corresponding Let/Set because these constitute one unique point of access on the interface
+            return _members.Count() > ExcessiveMemberCount;
+        }
+
+        private static bool NoMatchingSetter(Declaration property) 
+        {
+            var setter = _members.Where(member => member.IdentifierName == property.IdentifierName);
+            return setter.Count() == 0; //if setter count is 1 then there is a matching setter so this returns false
```

Comment:
This condition might work, but the comment is misleading since there might be a Set and a Let.
Moreover, you should not use `Where` and then `Count` here. This is a use case for `Any`.